craft_parts/utils/partition_utils.py

```diff
@@ -118,58 +115,123 @@ def _validate_partition_naming_convention(partitions: Sequence[str]) -> None:
                 ),
             )
 
+        if _is_valid_partition_name(partition):
+            continue
         raise errors.FeatureError(
             message=f"Partition {partition!r} is invalid.",
             details=PARTITION_INVALID_MSG,
         )
 
 
-def _validate_namespace_conflicts(partitions: Sequence[str]) -> None:
-    """Validate conflicts between regular partitions and namespaces.
+def _validate_partitions_conflicts(partitions: Sequence[str]) -> None:
+    """Validate conflicts between partitions.
+
+    Assumes partition names are valid and unique.
+
+    :raises FeatureError: for conflicts
+    """
+    conflictings_partitions = detect_conflicts(partitions)
+
+    if not conflictings_partitions:
+        return
+
+    filtered_conflictings_partitions = unique_conflicts(conflictings_partitions)
+
+    # Raise the full list of collisions
+    lines: list[str] = []
+    for conflictings in filtered_conflictings_partitions.values():
```

Comment:
Shouldn't that be `conflicting` without the `s`? Maybe `for conflict in filtered_conflicting_partitions`?